config/config.go

```diff
@@ -268,3 +276,25 @@ func ValidateEmail(email string) bool {
 	Re := regexp.MustCompile(`^[a-z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]{2,4}$`)
 	return Re.MatchString(email)
 }
+
+func ValidateEndpoints(daemonEndpoint string, passthroughEndpoint string) error {
+	passthroughURL, err := url.Parse(passthroughEndpoint)
+	if err != nil {
+		return errors.New("passthrough endpoint must be a valid URL")
+	}
+	daemonHost, daemonPort, err := net.SplitHostPort(daemonEndpoint)
+	if err != nil {
+		return errors.New("couldn't split host:post of daemon endpoint")
+	}
+
+	if daemonHost == passthroughURL.Hostname() && daemonPort == passthroughURL.Port() {
+		return errors.New("passthrough endpoint can't be the same as daemon endpoint!")
+	}
+
+	if ((daemonPort == passthroughURL.Port()) &&
+	    (daemonHost == "0.0.0.0") &&
```

Comment:
we will have a problem when say daemonHost="localhost"  and daemonHost != passThroghHost no ?

Reply:
I don't understand what problem you are thinking of. Could you give an example of the two endpoint strings, and what you'd expect to have happen?

I have added tests so it's clear what values should or shouldn't work. Should be easy to add a new one if you want to confirm your example.

Reply:
Apologies should  have been more clear ,  the example below will also be a case where daemon is calling itself  no? 
  "passthrough_endpoint": "http://0.0.0.0:7003",
  "daemon_end_point": "localhost:7003",

Reply:
I am good, @Dinesh2521  if you dont have any further comments , we cal close this PR today itself